Fix #1141 : Removed error icons from password so that visibility toggle is clickable

Acash512 opened the pull request:
### Description
Resolved the bug that caused overlapping of error icon with visibility toggle in password editTexts for SignUpActivity

Fixes #1141 

#### Before
| During Runtime | After SignUp is clicked |
|---|---|
| <img src = "https://user-images.githubusercontent.com/53511962/120353013-f7471c80-c31e-11eb-9db2-51a47933f3c4.jpeg" width=350/> | <img src = "https://user-images.githubusercontent.com/53511962/120353039-fc0bd080-c31e-11eb-9fe7-c83103c1391f.jpeg" width=350/> |

#### Now
| During Runtime | After SignUp is clicked |
|---|---| 
| <img src = "https://user-images.githubusercontent.com/53511962/120842670-a50a3380-c58a-11eb-9061-8d5900005e40.jpeg" width=350/> | <img src = "https://user-images.githubusercontent.com/53511962/120842728-b6534000-c58a-11eb-914c-02297e60445b.jpeg" width=350/> |

### Type of Change:

- Code
- User Interface

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)

### Checklist:
<!--- **Delete irrelevant options.** --->

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings

Acash512 committed:
Removed error icons from password so that visibility toggle is clickable

epicadk commented:
Please post a screen shot of the changes

Acash512 commented:
> Please post a screen shot of the changes

Done

@epicadk I also noticed just now that we have checked the format of email, only during runtime, and not when signup button is clicked, earlier since the error was set on tiEmail.editText, it was retained even when Sign Up button was clicked, but now it isn't as I have the set the error on tiEmail during runtime(which is set null in validateDetails funcn invoked when Sign Up is clicked as email is not blank and it's format is not checked there) so that UI looks similar.

So shall I set the the error on tiEmail.editText as before during runtime or check if the email has correct format or not when the Sign Up button is clicked too and set the error accordingly.

Please take a look into this too, I will accordingly update the PR as per your suggestion, sorry for bothering you with this now, I should have noticed that earlier.

epicadk commented:
@isabelcosta please approve pr run.

> > Please post a screen shot of the changes
> 
> Done
> 
> @epicadk I also noticed just now that we have checked the format of email, only during runtime, and not when signup button is clicked, earlier since the error was set on tiEmail.editText, it was retained even when Sign Up button was clicked, but now it isn't as I have the set the error on tiEmail during runtime(which is set null in validateDetails funcn invoked when Sign Up is clicked as email is not blank and it's format is not checked there) so that UI looks similar.
> 
> So shall I set the the error on tiEmail.editText as before during runtime or check if the email has correct format or not when the Sign Up button is clicked too and set the error accordingly.
> 
> Please take a look into this too, I will accordingly update the PR as per your suggestion, sorry for bothering you with this now, I should have noticed that earlier.

Sure you can check the it on the signup click event.

epicadk commented:
@isabelcosta please approve pr run.

epicadk commented on app/src/main/res/layout/activity_sign_up.xml:
```diff
@@ -106,6 +107,7 @@
             android:layout_height="wrap_content"
             android:layout_marginTop="16dp"
             android:hint="@string/confirm_password"
+            app:errorIconDrawable="@null"
```
Remove this line?

epicadk commented on app/src/main/res/layout/activity_sign_up.xml:
```diff
@@ -88,6 +88,7 @@
             android:layout_height="wrap_content"
             android:layout_marginTop="16dp"
             android:hint="@string/password"
+            app:errorIconDrawable="@null"
```
Remove this line?

Acash512 committed:
Made recommended changes

Acash512 committed:
Resolved few issues

Acash512 commented:
> @isabelcosta please approve pr run.
> 
> > > Please post a screen shot of the changes
> > 
> > 
> > Done
> > @epicadk I also noticed just now that we have checked the format of email, only during runtime, and not when signup button is clicked, earlier since the error was set on tiEmail.editText, it was retained even when Sign Up button was clicked, but now it isn't as I have the set the error on tiEmail during runtime(which is set null in validateDetails funcn invoked when Sign Up is clicked as email is not blank and it's format is not checked there) so that UI looks similar.
> > So shall I set the the error on tiEmail.editText as before during runtime or check if the email has correct format or not when the Sign Up button is clicked too and set the error accordingly.
> > Please take a look into this too, I will accordingly update the PR as per your suggestion, sorry for bothering you with this now, I should have noticed that earlier.
> 
> Sure you can check the it on the signup click event.



> @isabelcosta please approve pr run.
> 
> > > Please post a screen shot of the changes
> > 
> > 
> > Done
> > @epicadk I also noticed just now that we have checked the format of email, only during runtime, and not when signup button is clicked, earlier since the error was set on tiEmail.editText, it was retained even when Sign Up button was clicked, but now it isn't as I have the set the error on tiEmail during runtime(which is set null in validateDetails funcn invoked when Sign Up is clicked as email is not blank and it's format is not checked there) so that UI looks similar.
> > So shall I set the the error on tiEmail.editText as before during runtime or check if the email has correct format or not when the Sign Up button is clicked too and set the error accordingly.
> > Please take a look into this too, I will accordingly update the PR as per your suggestion, sorry for bothering you with this now, I should have noticed that earlier.
> 
> Sure you can check the it on the signup click event.

@epicadk I have made changes accordingly please review, I have set error on email for incorrect format even when sign up is clicked, and other changes that I have added in the third commit were to make sure that the error is set null during runtime when the input becomes valid, as far as the line that you mentioned is concerned I added it to ensure that visibility toggle is not covered by error icon in confirm password.

Acash512 committed:
Resolved Conflicts

Acash512 commented:
Resolved merge conflicts, there were some formatting issues

epicadk commented:
@isabelcosta you'll probably need to approve this again.

gaurivn reviewed (approved):
LGTM 👍, Thank you

isabelcosta committed:
Merge branch 'develop' into VisibilityToggle

isabelcosta reviewed (approved):
Nice work @Acash512 🙌🏾 Thank you for contributing!

vj-codes reviewed (approved):
_The changes made in this PR were tested locally. Following are the results_:

1. Code review - Done
2. All possible responses (positive and negative tests) were tested as below:
 
**Before:** 
Error icons from password used to overlap the visibility toggle making it difficult to click
**After:** 
Error icons from password have been removed and displayed below the field so that visibility toggle is clickable 

3. Status of PR : Ready to merge 🎉 
 

https://user-images.githubusercontent.com/60894542/134718182-4c0bb8d3-c1ab-40bf-9864-46624c54a480.mp4

isabelcosta commented:
Looks good to me :) Thank you so much @vj-codes for testing this PR 🥳 🙌🏾 
Note: Vaishnavi and I noticed there is a little bug where the password mismatch text changes but this was not created through this PR, so an issue to fix this bug will follow up :)
